src/binary/Builtin.cc

```diff
@@ -632,6 +632,61 @@ extern "C" {
     return INT2FIX(strcoll(RSTRING_PTR(str1), RSTRING_PTR(str2)));
   }
 
+  // reads the value of a hash key
+  // used internally by stftime_wrapper
+  static int
+  tm_hash_value(VALUE hash, const char *key)
+  {
+    return NUM2INT(rb_hash_aref(hash, ID2SYM(rb_intern(key))));
+  }
+  
+  // converts a hash to a tm structure
+  // used internally by stftime_wrapper
+  void
+  hash_to_tm(VALUE hash, struct tm *res)
+  {
+    res->tm_sec   = tm_hash_value(hash, "tm_sec");
+    res->tm_min   = tm_hash_value(hash, "tm_min");
+    res->tm_hour  = tm_hash_value(hash, "tm_hour");
+    res->tm_mday  = tm_hash_value(hash, "tm_mday");
+    res->tm_mon   = tm_hash_value(hash, "tm_mon");
+    res->tm_year  = tm_hash_value(hash, "tm_year");
+    res->tm_wday  = tm_hash_value(hash, "tm_wday");
+    res->tm_yday  = tm_hash_value(hash, "tm_yday");
+    res->tm_isdst = tm_hash_value(hash, "tm_isdst");
+  }
+
+#define STRFTIME_MAX_LENGTH 256
+  // a wrapper around strftime() function,
+  // needed for formatting using the current locale
+  static VALUE
+  strftime_wrapper(VALUE self, VALUE time, VALUE format)
+  {
+    char res[STRFTIME_MAX_LENGTH];
+    struct tm timeinfo;
+
+    Check_Type(format, T_STRING);
+    hash_to_tm(time, &timeinfo);
+
+    try
+    {
+      // Since std::put_time is not implemented in GCC4.9, we'll use
+      // locale::global and strftime instead of facets
+      // FIXME: I'm not sure about the implications!!!!
```

Comment:
This will effect all C++ libraries linked to YaST. On the other hand setting the locale it is the right thing to do anyway.